PowerShellTools.HostService/ServiceManagement/Debugging/PowershellDebuggingServiceUtilities.cs

```diff
@@ -145,28 +147,48 @@ private void InitializeRunspace(PSHost psHost)
             _runspace = RunspaceFactory.CreateRunspace(psHost, iss);
             _runspace.Open();
 
-            LoadProfile();
+            ProvideProfileVariable();
+            LoadProfiles();
             ServiceCommon.Log("Done initializing runspace");
         }
 
-        private void LoadProfile()
+        private void LoadProfiles()
         {
-            ServiceCommon.Log("Loading PowerShell Profile");
+            if (_callback == null && OperationContext.Current != null)
+            {
+                _callback = OperationContext.Current.GetCallbackChannel<IDebugEngineCallback>();
+            }
+
+            if (_callback == null || !_callback.ShouldLoadProfiles())
+            {
+                // if we can't reach the _callback on startup, let's not assume anything and skip loading profiles
+                return;
+            }
+
+            ServiceCommon.Log("Loading PowerShell Profiles");
             using (PowerShell ps = PowerShell.Create())
             {
-                var myDocuments = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);
-                var windowsPowerShell = Path.Combine(myDocuments, "WindowsPowerShell");
-                var profile = Path.Combine(windowsPowerShell, "PoshTools_profile.ps1");
+                ps.Runspace = _runspace;
 
-                var fi = new FileInfo(profile);
-                if (!fi.Exists)
+                PSObject profiles = _runspace.SessionStateProxy.PSVariable.Get("profile").Value as PSObject;
+                foreach(PSMemberInfo member in profiles.Members)
                 {
-                    return;
+                    if (member.MemberType == PSMemberTypes.NoteProperty)
+                    {
+                        var profilePath = (string)member.Value;
+                        var profileFile = new FileInfo(profilePath);
+
+                        if (!profileFile.Exists)
+                        {
+                            return;
+                        }
+
+                        ServiceCommon.Log(string.Format("Profile of name {0} found at {1}.", member.Name, profilePath));
+                        ps.Commands.Clear();
+                        ps.AddScript(". '" + profilePath + "'");

```

Comment:
That line was copy and pasted from code Adam had already written. I basically rewrote this method to load multiple profiles instead of just one. I'll change it to string.format though.